biom/table.py

```diff
@@ -948,6 +948,65 @@ def _get_col(self, col_idx):
         self._data = self._data.tocsc()
         return self._data.getcol(col_idx)
 
+    def align_metadata(self, metadata, axis='sample'):
```

Comment:
Would it be possible to rename this as `align_to_dataframe`? The `Table` already has a notion of `metadata` and what's being used here is external information rather than the metadata members